lib/CPathfinder.cpp

```diff
@@ -479,25 +558,24 @@ CPathsInfo::~CPathsInfo()
 	delete [] nodes;
 }
 
-const CGPathNode * CPathsInfo::getPathInfo( int3 tile ) const
+const CGPathNode * CPathsInfo::getPathInfo(const int3 &tile, const EPathfindingLayer &layer) const
 {
 	boost::unique_lock<boost::mutex> pathLock(pathMx);
 
-	if(tile.x >= sizes.x || tile.y >= sizes.y || tile.z >= sizes.z)
+	if(tile.x >= sizes.x || tile.y >= sizes.y || tile.z >= sizes.z || layer >= EPathfindingLayer::NUM_LAYERS)
```

Comment:
Actually I'm not sure if explicit assertion needed if I'm going to use multi_array as it's have built-in boundary checking. Though boundary checking is clearly hurt performance that is important for pathfinder so likely it's will need to be disabled.